### PR TITLE
MDEXP-259 Log errors for Holding and Item records while export

### PR DESCRIPTION
**Pull request by andrii-novytskyi**
## Purpose
https://issues.folio.org/browse/MDEXP-259

## Approach
Added error logs while mapping for holdings and items;
Added error logs in Clients;
Modified unit tests;
Reduced number of sonar issues

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.

- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [x] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/main/java/org/folio/clients/ClientException.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package org.folio.clients;
+
```
Comment (Igor-Gorchakov) [inline src/main/java/org/folio/clients/ClientException.java:2]: Javadoc

File: src/main/java/org/folio/clients/ClientException.java [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+public class ClientException extends Exception {
```
Comment (Igor-Gorchakov) [inline src/main/java/org/folio/clients/ClientException.java:3]: Add java doc please

**Review by Igor-Gorchakov — APPROVED** [review]

File: src/main/java/org/folio/clients/InventoryClient.java [hunk 3]
```diff
@@ -145,20 +175,26 @@
...
-        QUERY_PATTERN_ITEM);
-  }
-
-  public Map<String, JsonObject> getContributorNameTypes(OkapiConnectionParams params) {
-    String endpoint = resourcesPathWithPrefix(CONTRIBUTOR_NAME_TYPES) + LIMIT_PARAMETER + REFERENCE_DATA_LIMIT;
-    return getReferenceDataByUrl(endpoint, params, CONTRIBUTOR_NAME_TYPES);
+    try {
+      return getRequest(params, endpoint);
+    } catch (ClientException exception) {
+      errorLogService.saveGeneralError("Error while getting holdings by instance ids " + exception.getMessage(), jobExecutionId, params.getTenantId());
```
Comment (KVupp) [inline src/main/java/org/folio/clients/InventoryClient.java:184]: Log the instance Id, so that we know for which instance we couldn't get the holdings

Comment (KVupp) [reply src/main/java/org/folio/clients/InventoryClient.java:184]: ok nvm, the details should be there in the exception message thrown?

Comment (andrii-novytskyi) [reply src/main/java/org/folio/clients/InventoryClient.java:184]: will add

File: src/main/java/org/folio/clients/ClientException.java (hunk 2 above)
Comment (andrii-novytskyi) [reply src/main/java/org/folio/clients/ClientException.java]: Added

**Review by KVupp — COMMENTED** [review]
Everything else LGTM, please just add an integration test . So that we  can test the end-end flow of fetching the Error logs using the job executionId

**Comment by andrii-novytskyi** [comment]
@KVupp already added it in the last commit

File: src/main/java/org/folio/clients/ClientUtil.java [hunk 4]
```diff
@@ -88,7 +87,7 @@ public static JsonObject getResponseEntity(CloseableHttpResponse response) {
         LOGGER.error("Exception while building response entity", e);
       }
     }
-    return null;
+    throw new IOException("Get invalid response with status: " + response.getStatusLine().getStatusCode());
```
Comment (KVupp) [inline src/main/java/org/folio/clients/ClientUtil.java:90]: I'm guessing we need to be more clear with what the exception is. We can go ahead and merge it for now, but while testing if we see any issues, let's come back to improve this

**Review by KVupp — APPROVED** [review]

**Review by KVupp — APPROVED** [review]

**Pull request merged**